src/main/java/cd/go/plugin/config/yaml/JsonnetConfigParser.java

```diff
@@ -0,0 +1,76 @@
+package cd.go.plugin.config.yaml;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import cd.go.plugin.config.yaml.transforms.RootTransform;
+
+public class JsonnetConfigParser extends YamlConfigParser {
+    public JsonnetConfigParser() {
+        super(new RootTransform());
+    }
+
+    @Override
+    public JsonConfigCollection parseFiles(File baseDir, String[] files) {
+        JsonConfigCollection collection = new JsonConfigCollection();
+        for (String file : files) {
+            try {
+                File filePath = new File(baseDir, file);
+                InputStream input = compileJsonnet(filePath.toString());
+                // Calling YamlConfigParser's parseStream method (instead of the overridden one below)
+                super.parseStream(collection, input, file);
+            } catch (NullPointerException e) {
+                collection.addError("File matching GoCD Jsonnet pattern disappeared", file);
+            }
+        }
+        return collection;   
+    }
+
+    @Override
+    public void parseStream(JsonConfigCollection result, InputStream input, String location) {
+        try {
+            InputStream jsonInputStream = compileJsonnet(input);
+            super.parseStream(result, jsonInputStream, location);
+        } catch (NullPointerException ex) {
+            result.addError("File matching GoCD Jsonnet pattern disappeared", location);
+        }
```

Comment:
errors need to result in result.addError, rather than a print.

```suggestion
        } catch (JsonnetEvalException ex) {
            result.addError(new PluginError(
                                String.format("Failed to evaluate jsonnet: %s", ex.getMessage()), location));
        }
```
to create the custom exception, see 
https://github.com/tomzo/gocd-yaml-config-plugin/blob/master/src/main/java/cd/go/plugin/config/yaml/YamlConfigException.java